lib/ansible/modules/windows/win_secedit.ps1

```diff
@@ -0,0 +1,179 @@
+    )
+
+        $ini = New-Object System.Collections.Specialized.OrderedDictionary
+        $currentSection = New-Object System.Collections.Specialized.OrderedDictionary
+        $curSectionName = "default"
+
+        switch -regex (gc $FilePath)
+        {
+            "^\[(?<Section>.*)\]"
+            {
+                        $ini.Add($curSectionName, $currentSection)
+
+                        $curSectionName = $Matches['Section'].trim()
+                        $currentSection = New-Object System.Collections.Specialized.OrderedDictionary
+            }
+                "(?<Key>.*)\=(?<Value>.*)"
+                {
+                        # add to current section Hash Set
+                        $currentSection.Add($Matches['Key'].trim(), $Matches['Value'].trim())
+                }
+                "^$"
+                {
+                        # ignore blank line
+                }
+                default
+                {
+                        throw "Unidentified: $_"  # should not happen
+                }
+        }
+        if ($ini.Keys -notcontains $curSectionName) { $ini.Add($curSectionName, $currentSection) }
+
+        return $ini
+}
+
+function Out-IniFile{
+    param (
+        [parameter(mandatory=$true, position=0, valuefrompipelinebypropertyname=$true, valuefrompipeline=$true)][System.Collections.Specialized.OrderedDictionary]$ini,
+                [parameter(mandatory=$false,position=1, valuefrompipelinebypropertyname=$true, valuefrompipeline=$false)][String]$FilePath
+    )
+
+        $output = ""
+        ForEach ($section in $ini.GetEnumerator())
+        {
+                if ($section.Name -ne "default")
+                {
+                        # insert a blank line after a section
+                        $sep = @{$true="";$false="`r`n"}[[String]::IsNullOrWhiteSpace($output)]
+                        $output += "$sep[$($section.Name)]`r`n"
+                }
+                ForEach ($entry in $section.Value.GetEnumerator())
+                {
+                        $sep = @{$true="";$false="="}[$entry.Name -eq ";"]
+                        $output += "$($entry.Name)$sep$($entry.Value)`r`n"
+                }
+
+        }
+
+        $output = $output.TrimEnd("`r`n")
+        if ([String]::IsNullOrEmpty($FilePath))
+        {
+                return $output
+        }
+        else
+        {
+                $output | Out-File -FilePath $FilePath -Encoding:ASCII
+        }
+}
+
+$params = Parse-Args $args;
+
+$result = New-Object psobject @{
+    changed = $false
+};
+
+$category = Get-Attr $params "category" -failifempty $true
+$key = Get-Attr $params "key" -failifempty $true
+$value = Get-Attr $params "value" -failifempty $true
+$sepath = "$home\sec_edit_dump.inf"
+
+If ((Get-WmiObject Win32_ComputerSystem).PartOfDomain) {
+    Fail-Json $result "This host is joined to a Domain Controller, you'll need to modify GPO directly instead of secedit"
```

Comment:
I don't think this should be here, in some GPO policies I've seen set definitely allowed me to change certain entries just not others. This should probably be a warning in the python document.